External/FEXCore/include/FEXCore/Utils/EnumUtils.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+// Header for various utilities related to operating with enums
+
+#include <type_traits>
+
+namespace FEXCore {
+
+// Macro that defines all of the built in operators for conveniently using
+// enum classes as flag types without needing to define all of the basic
+// boilerplate.
+#define FEX_DECLARE_ENUM_FLAG_OPERATORS(type)                                                      \
+    [[nodiscard]] constexpr type operator|(type a, type b) noexcept {                              \
+        using T = std::underlying_type_t<type>;                                                    \
+        return static_cast<type>(static_cast<T>(a) | static_cast<T>(b));                           \
+    }                                                                                              \
+    [[nodiscard]] constexpr type operator&(type a, type b) noexcept {                              \
+        using T = std::underlying_type_t<type>;                                                    \
+        return static_cast<type>(static_cast<T>(a) & static_cast<T>(b));                           \
+    }                                                                                              \
+    [[nodiscard]] constexpr type operator^(type a, type b) noexcept {                              \
+        using T = std::underlying_type_t<type>;                                                    \
+        return static_cast<type>(static_cast<T>(a) ^ static_cast<T>(b));                           \
+    }                                                                                              \
+    [[nodiscard]] constexpr type operator<<(type a, type b) noexcept {                             \
```

Comment:
All operators in this file are useful to have, but having shift operators seems esoteric and I'd expect this to be used accidentally more often than intentionally. Are these used anywhere currently and/or do other projects use this commonly?

Without wide use I'd opt for requiring the user to do any such shifting explicitly instead.

What I could see more commonly used is an expression like `1 << ENUM_VALUE`, which wouldn't compile by the current definition (I hope? You never know with implicit conversions...). Perhaps a more useful alternative is `operator<<(underlying_int, type)` hence?

Reply:
These are currently unused, so we can just remove them for the time being.